module/lifecycle/lifecycle_test.go

```diff
@@ -25,9 +25,10 @@ func (suite *LifecycleManagerSuite) SetupTest() {
 func (suite *LifecycleManagerSuite) TestConcurrentStart() {
 	var numStarts uint32
 
+	lm := suite.lm
 	for i := 0; i < 10; i++ {
 		go func() {
-			suite.lm.OnStart(func() {
+			lm.OnStart(func() {
```

Comment:
Just wondering, is this an algorithmic change?

Reply:
I think this happens because `suite.lm` is shared between multiple test cases. 
The next test case will run the same`SetupTest` method, which will update `suite.lm`. However, that's a concurrent write, which is a race to a concurrent read here from `suite.lm.OnStart`.

The fix was to create a new value `lm`. So that the concurrent write from other test cases to `suite.lm` will not be a race to the read to `lm`.

In other words, it's purely a test case bug rather than an algorithmic change.